Google.Api.Gax.Grpc/Gcp/GcpCallInvokerPool.cs

```diff
@@ -26,28 +26,23 @@ public sealed class GcpCallInvokerPool
 
         private readonly DefaultChannelCredentialsCache _credentialsCache;
 
+        private readonly ServiceMetadata _serviceMetadata;
         private readonly Dictionary<Key, GcpCallInvoker> _callInvokers = new Dictionary<Key, GcpCallInvoker>();
         private readonly object _lock = new object();
 
+        // TODO: 
+
         /// <summary>
-        /// Creates a call invoker pool which will apply the specified scopes to the default application credentials
-        /// if they require any.
+        /// Creates a call invoker pool which will use the given service metadata to determine scopes
+        /// and self-signed JWT support.
         /// </summary>
-        /// <param name="scopes">The scopes to apply. Must not be null, and must not contain null references. May be empty.</param>
-        public GcpCallInvokerPool(IEnumerable<string> scopes) : this(scopes, false)
+        /// <param name="serviceMetadata"></param>
+        public GcpCallInvokerPool(ServiceMetadata serviceMetadata)
```

Comment:
This means that invokers cannot be shared across services in the same API, even for gRPC? Or am I missing something? Or were these per service before?

I think it would be OK to allow the per service granularity, but we probably want to keep the per API pools?

Reply:
OK, I see that at least the normal ChannelPool is per-service right now.